src/html5/parser.rs

```diff
@@ -2855,8 +2855,8 @@ impl<'stream> Html5Parser<'stream> {
 
                 self.pop_until("template");
                 self.active_formatting_elements_clear_until_marker();
-                self.template_insertion_mode.pop();
                 self.reset_insertion_mode();
+                self.template_insertion_mode.pop();
```

Comment:
previous modify cause html5-parse-test panic for template.dat, so revert the modify. I thought the problem might be template contents support.